plugins/builds/index.js

```diff
@@ -910,16 +953,46 @@ exports.register = (server, options, next) => {
                     const jobId = parentWorkflowGraph.nodes.find(node =>
                         node.name === trimJobName(externalJobName)).id;
                     // Get next build
-                    const nextBuild = finishedExternalBuilds.find(b => b.jobId === jobId);
+                    const nextBuild = finishedExternalBuilds.find(b => b.jobId === jobId
+                        && b.status === 'CREATED');
+                    const isRestartCase = finishedExternalBuilds.find(b => b.jobId === jobId
```

Comment:
if I understand this correctly, this is`isRestartCaseWithSuccessfulBuild` any other scenario this build won't exist and `nextBuild` will be in `CREATED` state

Reply:
the previous event could have had successful runs for the fork jobs and just failed during the join job